### PR TITLE
OSS more things

### PR DESCRIPTION
**Pull request by ash2k**

### CODE REVIEW

File: pkg/execution/computionadmission/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Compution
+
+## Admission Controller
```
Comment (nilebox) [inline pkg/execution/computionadmission/README.md]: This thing is Atlassian specific. Shall we move it to an `/atlassian/` subpackage?

File: pkg/execution/computionadmission/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+that denies compute resources if they reference non-compliant (i.e. not under /sox/ ) artifacts when the service requires PRGB compliance.
+
+## Controller (TODO)
+
+This monitors compute resources,
+and ensures that Service Central has a compute record for each compute resource.
+
+## See Also
+
+- https://hello.atlassian.net/wiki/spaces/VDEV/pages/325687047/ServiceCentralComputeRecord+Controller
```
Comment (nilebox) [inline pkg/execution/computionadmission/README.md]: These links are private

File: pkg/execution/secretive/asap/client.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import (
+	"os"
+	"time"
+
+	asap "bitbucket.org/atlassian/go-asap"
+	"bitbucket.org/atlassian/go-asap/keyprovider"
+	"github.com/pkg/errors"
+)
+
+type Client struct {
```
Comment (nilebox) [inline pkg/execution/secretive/asap/client.go]: This one is also Atlassian specific?

File: pkg/execution/svccatadmission/admission.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+	scClient := servicecentral.NewServiceCentralClient(s.Logger, scHTTPClient, s.ASAPClientConfig, s.ServiceCentralURL)
+
+	rpsHTTPClient := util.HTTPClient()
+	rpsCache := rps.NewRPSCache(s.Logger, rps.NewRPSClient(s.Logger, rpsHTTPClient, s.ASAPClientConfig, s.RPSURL))
+
+	r.Post("/externalid", admission.AdmitFuncHandlerFunc("externalid",
+		func(ctx context.Context, logger *zap.Logger, admissionReview admissionv1beta1.AdmissionReview) (*admissionv1beta1.AdmissionResponse, error) {
+			return ExternalUUIDAdmitFunc(ctx, uuid.Default(), scClient, rpsCache, admissionReview)
+		}))
+	r.Post("/micros", admission.AdmitFuncHandlerFunc("micros",
```
Comment (nilebox) [inline pkg/execution/svccatadmission/admission.go]: Atlassian-specific.

File: pkg/execution/svccatadmission/migration.go [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+	}
+
+	namespaceService, err := getServiceData(ctx, scClient, namespaceServiceName)
+	if err != nil {
+		return nil, err
+	}
+
+	if namespaceService == nil {
+		reason := fmt.Sprintf(
+			"namespace service %q does not exist in Service Central - should be impossible",
```
Comment (nilebox) [inline pkg/execution/svccatadmission/migration.go]: Atlassian specific

File: pkg/pkiutil/asap_client.go [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+	"bitbucket.org/atlassian/go-asap"
+	"github.com/SermoDigital/jose/jws"
+	"github.com/atlassian/voyager/pkg/util"
+	"github.com/atlassian/voyager/pkg/util/validation"
+	"github.com/pkg/errors"
+	"k8s.io/api/core/v1"
+)
+
+type ASAP interface {
```
Comment (nilebox) [inline pkg/pkiutil/asap_client.go]: Atlassian specific

File: pkg/pkiutil/asap_middleware.go [hunk 7]
```diff
@@ -0,0 +1,19 @@
+package pkiutil
+
+import (
+	"fmt"
+	"net/http"
+
+	"bitbucket.org/atlassianlabs/restclient"
+)
+
+// SSAM API Documentation - https://ssam.office.atlassian.com/api/
```
Comment (nilebox) [inline pkg/pkiutil/asap_middleware.go]: Private link

File: pkg/servicecentral/client.go [hunk 8]
```diff
@@ -0,0 +1,391 @@
...
+// See the outdated API spec there https://stash.atlassian.com/projects/MICROS/repos/service-central/browse/docs/api.raml
+// Or the actual API code (Python) there https://stash.atlassian.com/projects/MICROS/repos/service-central/browse/service_central/api.py
+type Client struct {
+	logger     *zap.Logger
+	httpClient *http.Client
+	asap       pkiutil.ASAP
+	rm         *restclient.RequestMutator
+}
+
+func NewServiceCentralClient(logger *zap.Logger, httpClient *http.Client, asap pkiutil.ASAP, baseURL *url.URL) *Client {
```
Comment (nilebox) [inline pkg/servicecentral/client.go]: Atlassian

File: pkg/servicecentral/testdata/create_service_rsp.json [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+  "data": [
+	{
+	  "availability_objective": 99.0,
+	  "business_unit": "PaaS/Micros",
+	  "compliance_policy_id": null,
+	  "creation_timestamp": "2018-05-17T07:40:48.302606Z",
+	  "description": null,
+	  "hipchat_room_name": null,
+	  "last_modified_timestamp": "2018-05-17T07:40:48.302606Z",
+	  "last_modified_user": "LDAPUser:aczepil",
```
Comment (nilebox) [inline pkg/servicecentral/testdata/create_service_rsp.json]: Unnecessary details of internal service

File: pkg/servicecentral/types.go [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+}
+
+type miscData struct {
+	Key   string `json:"misc_key,omitempty"`
+	Value string `json:"misc_value,omitempty"`
+	// note that SC does have a `uuid` field for misc, but we intentionally exclude it
+	// if we do not exclude it, then updates to existing fields will not work
+}
+
+// Copied from https://stash.atlassian.com/projects/MICROS/repos/central/browse/pkg/central/model/service.go
```
Comment (nilebox) [inline pkg/servicecentral/types.go]: Private link

File: pkg/servicecentral/types.go [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+	Platform *string `json:"platform" sql:"platform"`
+
+	// Description of the service
+	Description *string `json:"description" sql:"description"`
+
+	// URL of the service - which doesn't make alot of sense if it is a multi region/environment service
+	ServiceURL *string `json:"servingURL" sql:"serving_url"`
+
+	// Name of the Hipchat / Stride room where you can reach the team that maintains the service
+	HipChatRoomName *string `json:"hipchatRoomName" sql:"hipchat_room_name"`
```
Comment (nilebox) [inline pkg/servicecentral/types.go]: hmmm

File: pkg/servicecentral/types.go [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+type ServiceData struct {
+	ServiceUUID        *string      `json:"service_uuid,omitempty"`
+	CreationTimestamp  *string      `json:"creation_timestamp,omitempty"`
+	ServiceName        string       `json:"service_name,omitempty"`
+	ServiceOwner       ServiceOwner `json:"service_owner,omitempty"`
+	ServiceTier        int          `json:"service_tier,omitempty"`
+	Tags               []string     `json:"tags,omitempty"`
+	Platform           string       `json:"platform,omitempty"`
+	Misc               []miscData   `json:"misc,omitempty"`
+	PagerDutyServiceID string       `json:"pagerduty_service_id,omitempty"`
```
Comment (nilebox) [inline pkg/servicecentral/types.go]: Not sure if this should be public? We will add OpsGenie later as well?

Comment (jokeyrhyme) [reply pkg/servicecentral/types.go]: I believe we'll just have the OpsGenie team name as an optional `string`, but best to double-check :)

**Pull request merged**